mlir/test/Target/LLVMIR/omptarget-llvm.mlir

```diff
@@ -263,12 +263,12 @@ llvm.func @_QPopenmp_target_use_dev_addr() {
 
 llvm.func @_QPopenmp_target_use_dev_addr_no_ptr() {
   %0 = llvm.mlir.constant(1 : i64) : i64
-  %a = llvm.alloca %0 x !llvm.ptr<i32> : (i64) -> !llvm.ptr<i32>
-  %map = omp.map_info var_ptr(%a : !llvm.ptr<i32>, i32)   map_clauses(tofrom) capture(ByRef) -> !llvm.ptr<i32> {name = ""}
-  omp.target_data  map_entries(%map : !llvm.ptr<i32>) use_device_addr(%a : !llvm.ptr<i32>)  {
-  ^bb0(%arg0: !llvm.ptr<i32>):
+  %a = llvm.alloca %0 x i32 : (i64) -> !llvm.ptr
```

Comment:
I had to change this test a bit to get the expected output. @TIFitis @agozillon

Reply:
It seems reasonable to me, but I am not all that familiar with use_device_addr or this specific test, so perhaps @TIFitis has a better idea, or maybe @raghavendhra who I think might have more experience with use_device_addr/ptr

Reply:
This is fine.

I added this test to make sure use_dev_addr works with scalar variables, which I think is preserved.